### ISSUE TITLE
Training LF Discriminative baseline

### ISSUE THREAD

**Issue by davidnvq** [davidnvq/visdial#6]
# Some notes
- The learning rate is fixed at `1e-3`. 
- Replaced code by annotated version

# Comet ml 
* [comet link](https://www.comet.ml/lightcv/visdial-baselines/2b256fbc54774b9c8d7761d5cd09eca6)

# Results after 10 epochs
* The metrics
![image](https://user-images.githubusercontent.com/30741785/57422648-e31db780-724b-11e9-8f38-56ee2a206bcd.png)
* On `training`: 
```
r@1: 0.5258178114891052
r@5: 0.8338583707809448
r@10: 0.9158499836921692
mean: 3.6884424686431885
mrr: 0.6626250147819519
```

* On `validation`:
```
# Evaluate LF discriminative on Evaluation
r@1: 0.4777616262435913
r@5: 0.7927325367927551
r@10: 0.8862887620925903
mean: 4.466327667236328
mrr: 0.6199477314949036
ndcg: 0.5399369597434998
```

# Results from the pretrained model provided by author
```
# Evaluate LF discriminative on Evaluation
r@1: 0.5356588959693909
r@5: 0.8363372087478638
r@10: 0.9167635440826416
mean: 3.653294563293457
mrr: 0.6695590615272522
ndcg: 0.5420510768890381
Writing ranks to /home/ubuntu/datasets/myvisdial/checkpoints/lf_disc/baseline/output/val_ranks.json
```
